be/src/storage/lake/gc.cpp

```diff
@@ -34,12 +34,17 @@ Status metadata_gc(std::string_view root_location, TabletManager* tablet_mgr) {
     }
 
     std::unordered_map<int64_t, std::vector<int64_t>> tablet_metadatas;
+    std::vector<string> txn_logs;
+    std::unordered_map<int64_t, std::vector<int64_t>> locked_tablet_metadatas;
 
     auto iter_st = fs->iterate_dir(std::string(root_location), [&](std::string_view name) {
         if (is_tablet_metadata(name)) {
             auto [tablet_id, version] = parse_tablet_metadata_filename(name);
             tablet_metadatas[tablet_id].emplace_back(version);
         }
+        if (is_txn_log(name)) {
+            txn_logs.emplace_back(name);
```

Comment:
Dangerous: you can not delete txn_id file because it may be used by load task

Reply:
`txn_log` won't be deleted here. The `txn_log` with metadata lock will be deleted in the release snapshots phase.